test/game_team_test.rb

```diff
@@ -100,4 +100,59 @@ def test_it_can_find_winninest_coach
 
   def test_it_can_find_worst_coach
   end
+
+  def test_find_by_returns_games
+    assert_kind_of Array, GameTeam.find_by(2012030221)
+  end
+
+  def test_game_team_shots_goals_count
+    Game.from_csv('./test/fixtures/games_20.csv')
+    # arr_games = Game.all[0..2]
+    game_team1 = mock
+    game_team2 = mock
+    game_team3 = mock
+    game_team4 = mock
+    game_team1.stubs(:game_id).returns(2016020527)
+    game_team2.stubs(:game_id).returns(123)
+    game_team3.stubs(:game_id).returns(456)
+    game_team4.stubs(:game_id).returns(456)
+    passed_array = [game_team1,game_team2,game_team3,game_team4]
+    assert_kind_of Array , GameTeam.game_team_shots_goals_count(passed_array)
+  end
+
+  def test_get_goal_shots_by_game_team
+    game_team1 = mock
+    game_team2 = mock
+    game_team3 = mock
+    game_team4 = mock
+    game_team1.stubs(:team_id).returns(123)
+    game_team2.stubs(:team_id).returns(123)
+    game_team3.stubs(:team_id).returns(456)
+    game_team4.stubs(:team_id).returns(456)
+    game_team1.stubs(:goals).returns(3)
+    game_team2.stubs(:goals).returns(3)
+    game_team3.stubs(:goals).returns(2)
+    game_team4.stubs(:goals).returns(2)
+    game_team1.stubs(:shots).returns(6)
+    game_team2.stubs(:shots).returns(6)
+    game_team3.stubs(:shots).returns(4)
+    game_team4.stubs(:shots).returns(4)
+    passed_array = [game_team1,game_team2,game_team3,game_team4]
+
```

Comment:
This is a great example of how to test with mocks and stubs!

Reply:
Agreed!